File: backend/src/main/java/edu/cit/amihan/medibook/record/dto/HealthRecordResponse.java
package edu.cit.amihan.medibook.record.dto;

import edu.cit.amihan.medibook.record.entity.HealthRecord;
import java.time.LocalDateTime;

public class HealthRecordResponse {

    private Long recordId;
    private Long appointmentId;
    private String doctorName;
    private String patientName;
    private String diagnosis;
    private String consultationNotes;
    private LocalDateTime recordedAt;

    public static HealthRecordResponse fromEntity(HealthRecord rec) {
        HealthRecordResponse r = new HealthRecordResponse();
        r.recordId = rec.getRecordId();
        r.appointmentId = rec.getAppointment().getAppointmentId();
        r.doctorName = rec.getDoctor().getFullName();
        r.patientName = rec.getPatient().getFullName();
        r.diagnosis = rec.getDiagnosis();
        r.consultationNotes = rec.getConsultationNotes();
        r.recordedAt = rec.getRecordedAt();
        return r;
    }

    public Long getRecordId() { return recordId; }
    public Long getAppointmentId() { return appointmentId; }
    public String getDoctorName() { return doctorName; }
    public String getPatientName() { return patientName; }
    public String getDiagnosis() { return diagnosis; }
    public String getConsultationNotes() { return consultationNotes; }
    public LocalDateTime getRecordedAt() { return recordedAt; }
}

File: mobile/app/src/main/java/edu/cit/amihan/medibook/feature/appointment/model/HealthRecordResponse.kt
package edu.cit.amihan.medibook.feature.appointment.model

import com.google.gson.annotations.SerializedName

data class FdaDrugSuggestion(
    @SerializedName("brandName") val brandName: String? = null,
    @SerializedName("genericName") val genericName: String? = null,
    @SerializedName("route") val route: String? = null,
    @SerializedName("indication") val indication: String? = null
)

data class HealthRecordResponse(
    @SerializedName("recordId") val recordId: Long = 0L,
    @SerializedName("appointmentId") val appointmentId: Long = 0L,
    @SerializedName("doctorName") val doctorName: String? = null,
    @SerializedName("patientName") val patientName: String? = null,
    @SerializedName("diagnosis") val diagnosis: String? = null,
    @SerializedName("consultationNotes") val consultationNotes: String? = null,
    @SerializedName("recordedAt") val recordedAt: String? = null,
    @SerializedName("fdaSuggestions") val fdaSuggestions: List<FdaDrugSuggestion>? = null
)
